src/component/auth-context.jsx

```diff
@@ -0,0 +1,47 @@
+import React, { createContext, useContext, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+const AuthContext = createContext();
+
+export const useAuth = () => {
+    return useContext(AuthContext);
+};
+
+export const AuthProvider = ({ children }) => {
+    const [isLoggedIn, setIsLoggedIn] = useState(false);
+    const [username, setUsername] = useState(null);
+
+    useEffect(() => {
+        const token = localStorage.getItem('loginToken') || sessionStorage.getItem('loginToken');
+        const storedUsername = localStorage.getItem('username') || sessionStorage.getItem('username');
+        setIsLoggedIn(!!token);
+        setUsername(storedUsername);
+    }, []);
+
+    const login = (token, username, rememberMe) => {
+        const storage = rememberMe ? localStorage : sessionStorage;
+        storage.setItem('loginToken', token);
+        storage.setItem('username', username);
```

Comment:
would this be email? considering that we are not using a username in the backend?

Reply:
I initially set it as a username before realizing we were not storing it. Right now, that variable is storing the first name, and it will say Welcome (first name) next to the logout button. I can change it to be more semantic

Reply:
Sounds good!